packages/styled-components/src/test/ssr.test.js

```diff
@@ -5,7 +5,8 @@
 import React from 'react';
 import { renderToString, renderToNodeStream } from 'react-dom/server';
 import stylisRTLPlugin from 'stylis-plugin-rtl';
-import ServerStyleSheet from '../models/ServerStyleSheet';
+import ofStream from 'stream-to-promise';
+import { ServerStyleSheet } from '../../dist/styled-components.esm';
```

Comment:
I’m actually not sure anymore whether this is a valid reproduction. But it could also hint at the root cause.

This bundle includes the global `GroupIDAllocator`, but the `styled` function that’s used in this test is imported separately by Jest: https://github.com/styled-components/styled-components/blob/4add697ac770634300f7775fc880882b5497bdf4/packages/styled-components/src/test/utils.js#L8

So none of the group IDs are carried over and will match 🤔

Reply:
Yeah that makes sense. I'm still curious why [my repro](https://github.com/StevenLangbroek/sc-v5-streaming-repro) behaves the way it does then...